napari/plugins/_builtins.py

```diff
@@ -27,3 +36,200 @@ def napari_get_reader(path: Union[str, List[str]]) -> ReaderFunction:
+    table = np.concatenate([indices, table], axis=1)
+
+    # write table to csv file
+    write_csv(path, table, column_names)
+    return path
+
+
+@napari_hook_implementation(trylast=True)
+def napari_get_writer(
+    path: str, layer_types: List[str]
+) -> Optional[WriterFunction]:
+    """Our internal fallback file writer at the end of the writer plugin chain.
+
+    This will create a new folder from the path and call
+    ``napari_write_<layer>`` for each layer using the ``layer.name`` variable
+    to modify the path such that the layers are written to unique files in the
+    folder. It will use the default builtin writer for each layer type.
+
+    Parameters
+    ----------
+    path : str
+        path to file/directory
+
+    Returns
+    -------
+    callable
+        function that accepts the path and a list of layer_data (where
+        layer_data is ``(data, meta, layer_type)``) and writes each layer.
+    """
+    # normally, a plugin would do some logic here to decide whether it supports
+    # the ``path`` extension and layer_types.  But because this is our builtin
+    # "last resort" implementation, we just immediately hand back the writer
+    # function, and let it throw an exception if it fails.
+    return write_layer_data_with_plugins
+
+
+def write_layer_data_with_plugins(
+    path: str,
+    layer_data: List[FullLayerData],
+    *,
+    plugin_name: Optional[str] = 'builtins',
+    plugin_manager=None,
+) -> List[str]:
+    """Write layer data out into a folder one layer at a time.
+
+    Call ``napari_write_<layer>`` for each layer using the ``layer.name``
+    variable to modify the path such that the layers are written to unique
+    files in the folder.
+
+    If ``plugin_name`` is ``None`` then we just directly call
+    ``plugin_manager.hook.napari_write_<layer>()`` which will loop through
+    implementations and stop when the first one returns a non-None result. The
+    order in which implementations are called can be changed with the
+    implementation sorter/disabler.
+
+    If ``plugin_name`` is provided, then we call the
+    ``napari_write_<layer_type>`` for that plugin, and if it fails we error.
+    By default, we restrict this function to using only napari ``builtins``
+    plugins.
+
+    Parameters
+    ----------
+    path : str
+        path to file/directory
+    layer_data : list of napari.types.LayerData
+        List of layer_data, where layer_data is ``(data, meta, layer_type)``.
+    plugin_name : str, optional
+        Name of the plugin to use for saving. If None then all plugins
+        corresponding to appropriate hook specification will be looped
+        through to find the first one that can save the data. By default,
+        only builtin napari implementations are used.
+    plugins.
+    plugin_manager : plugins.PluginManager, optional
+        Instance of a napari PluginManager.  by default the main napari
+        plugin_manager will be used.
+
+    Returns
```

Comment:
I know changes are still in progress but just flagging so as not to forget.
Doc string here needs update I think
```
    Returns
    -------
    path : str or None
        If data is successfully written, return the ``path`` that was written.
        Otherwise, if nothing was done, return ``None``.
    """
```